### PR TITLE
Add project sharing view templates

### LINKED ISSUE #235: Project: add sharing admin view templates
Needed:

- [x] Sharing token listing
  - [x] Base listing and delete
  - [x] Decide on listing metadata
  - [x] Show expired state 
- [x] Sharing token creation form and likely JS view
- [x] Remove/edit etc views
- [x] Multi select on versions doesn't work
- [x] Tune content for sharing types on the form view
- [x] Add navigation breadcrumbs to form/detail view

- Requires https://github.com/readthedocs/readthedocs-corporate/pull/1695

### PR DESCRIPTION
**Pull request by agjohnson**
This adds:

- Listing page
- Edit/detail page
- Delete modal
- Admin dashboard navigation item

https://github.com/readthedocs/ext-theme/assets/1140183/1b7477c0-c240-4458-b63c-d549858597b4

- Closes #235
- Requires https://github.com/readthedocs/readthedocs-corporate/pull/1695

### CODE REVIEW

**Comment by agjohnson** [comment]
There isn't any deep review here required. The screengrab is probably most helpful to look over to ensure everything looks mostly correct for an initial pass.

File: readthedocsext/theme/templates/projects/partials/edit/temporary_access_list.html [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+  {% trans "Remove share" as action_text %}
+  {% blocktrans trimmed asvar content_text with share=object.description %}
+    Remove share {{ share }}?
+  {% endblocktrans %}
+  {% include "includes/crud/remove_button.html" with id=object.pk form_url=form_url action_text=action_text context_text=context_text %}
+{% endblock list_item_right_buttons %}
+
+{% block list_item_icon %}
+  {% if object.access_type == "password" %}
+    <i class="fa-duotone fa-input-text icon" data-content="{{ object.get_access_type_display }}"></i>
```
Comment (stsewd) [inline readthedocsext/theme/templates/projects/partials/edit/temporary_access_list.html:58]: Not sure if I convinced on using this icon for a password, maybe a key is a more familiar icon for a password?

File: readthedocsext/theme/templates/projects/partials/edit/temporary_access_list.html [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+  {% elif object.access_type == "token" %}
+    <i class="fa-solid fa-link icon" data-content="{{ object.get_access_type_display }}"></i>
+  {% elif object.access_type == "http_header_token" %}
+    <i class="fa-solid fa-right-left-large icon" data-content="{{ object.get_access_type_display }}"></i>
```
Comment (stsewd) [inline readthedocsext/theme/templates/projects/partials/edit/temporary_access_list.html:62]: a www/globe icon for this one maybe? :thinking:

File: readthedocsext/theme/templates/projects/partials/edit/temporary_access_list.html [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+  </a>
+  <div class="sub header">
+    {% if object.allow_all %}
+      {% blocktrans trimmed %}
+        All versions
+      {% endblocktrans %}
+    {% else %}
+      {# TODO consider listing the actual versions here #}
+      {% blocktrans trimmed %}
+        Select versions
```
Comment (stsewd) [inline readthedocsext/theme/templates/projects/partials/edit/temporary_access_list.html:78]:
```suggestion
        Selected versions
```

"Select versions" feels like an action not sure, it's probably right :thinking:

**Review by stsewd — APPROVED** [review]
Nothing blocking here

File: readthedocsext/theme/templates/projects/partials/edit/temporary_access_list.html (hunk 1 above)
Comment (agjohnson) [reply readthedocsext/theme/templates/projects/partials/edit/temporary_access_list.html:58]: Unfortunately not, I was a little surprised and not really happy with the icons for this view. Two other options are:

- https://fontawesome.com/icons/input-pipe?f=classic&s=duotone
- https://fontawesome.com/icons/pen-field?f=classic&s=duotone

And neither really fit really well either. I guess at least `pen-field` looks like a password?

File: readthedocsext/theme/templates/projects/partials/edit/temporary_access_list.html (hunk 2 above)
Comment (agjohnson) [reply readthedocsext/theme/templates/projects/partials/edit/temporary_access_list.html:62]: I wanted to keep the icons fairly specific, but unfortunately there are no good options for "HTTP request and response" :smile:

Some other options that are at least adjacent to HTTP transactions:

- https://fontawesome.com/icons/code?f=classic&s=duotone
- https://fontawesome.com/icons/terminal?f=classic&s=duotone
- https://fontawesome.com/icons/webhook?f=classic&s=duotone

They all have their own issues of course, but maybe more related? I suppose one of the `code` or related icons would be my next choice?

File: readthedocsext/theme/templates/projects/partials/edit/temporary_access_list.html (hunk 3 above)
Comment (agjohnson) [reply readthedocsext/theme/templates/projects/partials/edit/temporary_access_list.html:78]: Yeah, honestly neither are a really good fit, I'd rather replace this with something listing the versions explicitly, or at least counting them ("5 versions"). I would use "select" over "selected", as the past tense version is more ambiguous about what we previously selected.

I was going to save this, but if I have to jump back in, maybe I'll just replace this text now. I think a simple count works fine.

File: readthedocsext/theme/templates/projects/partials/edit/temporary_access_list.html (hunk 2 above)
Comment (stsewd) [reply readthedocsext/theme/templates/projects/partials/edit/temporary_access_list.html:62]: I like the one for code :+1:

File: readthedocsext/theme/templates/projects/partials/edit/temporary_access_list.html (hunk 1 above)
Comment (stsewd) [reply readthedocsext/theme/templates/projects/partials/edit/temporary_access_list.html:58]: pen-field seems better, yeah

**Comment by agjohnson** [comment]
With tweaked icons and I replaced the version list text:

image

I think the version list text could be tweaked more, I'd like to eventually see "Versions latest, stable, and 3 more", but Django templates are so weak that this is impossible without application code or dropping translations entirely :disappointed:

**Pull request merged**